### ISSUE TITLE
'InAppBillingImplementation' does not contain a definition for 'HandleActivityResult'

### ISSUE THREAD

**Issue by jumsintheflow** [jamesmontemagno/InAppBillingPlugin#308]
If you are creating an issue for a BUG please fill out this information. If you are asking a question or requesting a feature you can delete the sections below. 

**Failure to fill out this information will result in this issue being closed.** If you post a full stack trace in a bug it will be closed, please post it to http://gist.github.com and then post the link here.

## Bug Information

Version Number of Plugin: 4.0.0
Device Tested On: Honor 10 Android10
Simulator Tested On:
Version of VS:  Community 2019
Version of Xamarin: 16.7.000.440
Versions of other things you are using:

### Expected Behavior
InAppBillingImplementation should contain HandleActivityResult
### Actual Behavior
'InAppBillingImplementation' does not contain a definition for 'HandleActivityResult'

### Code snippet

protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            base.OnActivityResult(requestCode, resultCode, data);
            Plugin.InAppBilling.InAppBillingImplementation.HandleActivityResult(requestCode, resultCode, data);

        }

### Screenshotst

![Does not contain handleactivityresult](https://user-images.githubusercontent.com/13974473/92128486-f87c0500-edf1-11ea-9f0d-86c295e3bffd.png)

**Comment by jamesmontemagno** [comment]
Yes, it was removed, I have updated the readme to reflect this. 4.0 has major changes so please read.

**Comment by jumsintheflow** [comment]
Hi, thank you for your reply. i need further help please. my app crashes after payment, or even after hitting back button to exit the Google Play payment prompt . ==> System.NullReferenceException: 'Object reference not set to an instance of an object.'
i thought its because of the absence of HandleActivityResult. 
i've added this OnResume() but to no avail.
protected override void OnResume(){  base.OnResume();  }

**Comment by jumsintheflow** [comment]
try
            {
                productid = subscriptiones;
                connected = await CrossInAppBilling.Current.ConnectAsync();

                if (!connected)
                {
                    //Couldn't connect to billing
                    new AlertDialog.Builder(this)
                            .SetPositiveButton("OK", (sender2, args) => { })
                            .SetMessage("\n" + "Couldn't connect to Google Play billing," + "\n")
                            .SetTitle("Error!")
                            .Show();
                    return;
                }                                
                //try to purchase item
                verify = new Verify();
                purchase = await CrossInAppBilling.Current.PurchaseAsync(productid, ItemType.Subscription, verify);
                Toast.MakeText(ApplicationContext, "test test", ToastLength.Long).Show();

                if (purchase == null)
                {
                    //Not purchased, may also throw exception to catch                    
                    Toast.MakeText(ApplicationContext, "test null", ToastLength.Long).Show();
                }
                else
                {
                    //Purchased, save this information                    
                    Toast.MakeText(ApplicationContext, "test not null", ToastLength.Long).Show();
                }
            }
            catch (Plugin.InAppBilling.InAppBillingPurchaseException purchaseEx)
            {
                Console.WriteLine("Issue: " + purchaseEx);
                var message = string.Empty;
                switch (purchaseEx.PurchaseError)
                {
                    case PurchaseError.AppStoreUnavailable:
                        message = "Currently the Google Play store seems to be unavailble. Try again later.";
                        Toast.MakeText(ApplicationContext, message, ToastLength.Long).Show();
                        break;
                    case PurchaseError.BillingUnavailable:
                        message = "Billing seems to be unavailable, please try again later.";
                        Toast.MakeText(ApplicationContext, message, ToastLength.Long).Show();
                        break;
                    case PurchaseError.PaymentInvalid:
                        message = "Payment seems to be invalid, please try again.";
                        Toast.MakeText(ApplicationContext, message, ToastLength.Long).Show();
                        break;
                    case PurchaseError.PaymentNotAllowed:
                        message = "Payment does not seem to be enabled/allowed, please try again.";
                        Toast.MakeText(ApplicationContext, message, ToastLength.Long).Show();
                        break;
                    default:
                        Toast.MakeText(ApplicationContext, purchaseEx.InnerException.ToString(), ToastLength.Long).Show();
                        break;
                }                

                //Decide if it is an error we care about
                if (string.IsNullOrWhiteSpace(message))
                    return;
            }
            catch (Exception ex)
            {
                Console.WriteLine("Issue connecting: " + ex);
            }
            finally
            {
                //Disconnect, it is okay if we never connected, this will never throw an exception
                await CrossInAppBilling.Current.DisconnectAsync();
            }

**Comment by jamesmontemagno** [comment]
I would suggest using 2.0 for now until 4.0 is out of beta and the docs have been updated.

**Comment by jumsintheflow** [comment]
OK. thank you
